aioshelly/block_device/coap.py

```diff
@@ -111,14 +114,25 @@ def _read_extended_field_value(value: int, raw_data: bytes) -> tuple[int, bytes]
         raise InvalidMessage("Option contained partial payload marker.")
 
 
-def socket_init(socket_port: int) -> socket.socket:
+def socket_init(
+    socket_ips: list[IPv4Address] | None = None,
+    socket_port: int = DEFAULT_COAP_PORT,
+) -> socket.socket:
     """Init UDP socket to send/receive data with Shelly devices."""
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.bind(("", socket_port))
-    _LOGGER.debug("Socket initialized on port %s", socket_port)
-    mreq = struct.pack("=4sl", socket.inet_aton("224.0.1.187"), socket.INADDR_ANY)
-    sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
+    if socket_ips:
+        mreq = socket.inet_aton("224.0.1.187")
+        for address in socket_ips:
+            _LOGGER.debug("Socket initialized on %s:%s", address, socket_port)
+            mreq += socket.inet_aton(address.exploded)
```

Comment:
This looks like its adding the ip addresses together?

Reply:
I think https://github.com/python-zeroconf/python-zeroconf/blob/4877829e6442de5426db152d11827b1ba85dbf59/src/zeroconf/_utils/net.py#L261 might be a good reference for how this should work